lib/checksizeof.cpp

```diff
@@ -479,3 +479,98 @@ void CheckSizeof::arithOperationsOnVoidPointerError(const Token* tok, const std:
     const std::string verbose = message + " Arithmetic operations on 'void *' is a GNU C extension, which defines the 'sizeof(void)' to be 1.";
     reportError(tok, Severity::portability, "arithOperationsOnVoidPointer", "$symbol:" + varname + '\n' + message + '\n' + verbose, CWE467, Certainty::normal);
 }
+
+//checkAllocationSizeStart--------------------------------------------------------------------------------------------------
+std::string getTypeVarName(enum ValueType::Type var_types)
+{
+    switch (var_types)
+    {
+    case ValueType::UNKNOWN_TYPE: return "UNKNOWN_TYPE";
+    case ValueType::NONSTD: return "NONSTD";
+    case ValueType::SMART_POINTER: return "SMART_POINTER";
+    case ValueType::CONTAINER: return "CONTAINER";
+    case ValueType::ITERATOR: return "ITERATOR";
+    case ValueType::VOID: return "void";
+    case ValueType::POD: return "POD";
+    case ValueType::RECORD: return "RECORD";
+    case ValueType::BOOL: return "bool";
+    case ValueType::CHAR: return "char";
+    case ValueType::SHORT: return "short";
+    case ValueType::WCHAR_T: return "WCHAR_T";
+    case ValueType::INT: return "int";
+    case ValueType::LONG: return "long";
+    case ValueType::LONGLONG: return "longlong";
+    case ValueType::UNKNOWN_INT: return "UNKNOWN_INT";
+    case ValueType::FLOAT: return "float";
+    case ValueType::DOUBLE: return "double";
+    case ValueType::LONGDOUBLE: return "longDouble";
+    }
+    return "UNKNOWN_INT";
+}
+
+const int getSizeOfType(enum ValueType::Type var_types)
```

Comment:
We already have a function in valueflow.cpp to get the size of the type:

https://github.com/danmar/cppcheck/blob/main/lib/valueflow.cpp#L1088